pkg/controllers/terminating/v1alpha1/controller.go

```diff
@@ -65,16 +66,27 @@ func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface
 
 // Reconcile executes a reallocation control loop for the resource
 func (c *Controller) Reconcile(ctx context.Context, object client.Object) (reconcile.Result, error) {
+	n := object.(*v1.Node)
+	if n.DeletionTimestamp == nil || !node.IsKarpenterProvisioned(n) {
+		return reconcile.Result{}, nil
+	}
 	// 1. Cordon terminable nodes
-	if err := c.terminator.cordonNodes(ctx); err != nil {
-		return reconcile.Result{}, fmt.Errorf("cordoning terminable nodes, %w", err)
+	if err := c.terminator.cordonNode(ctx, n); err != nil {
+		return reconcile.Result{}, fmt.Errorf("cordoning node %s, %w", n.Name, err)
+	}
+
+	// 2. Drain terminable nodes
+	drained, err := c.terminator.drainNode(ctx, n)
```

Comment:
When we discussed this, did we decide we were going to hold off on draining until the next reconcile loop, otherwise new pods could schedule during draining.

Reply:
Our discussion was with the assumption that we would patch the spec at the higher level `GenericController`, which we recently made the realization that we only want to patch the Status here. 

Since we are patching the node here, not the status, we want to cordon and patch this node before continuing on to drain.